.github/workflows/main.yml

```diff
@@ -3,10 +3,25 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches:
+      - main
     tags:
       - '*.*.*'
 
 jobs:
+  test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Check out the repo
+        uses: actions/checkout@v4
+      - 
+        name: Unit testing 
+        uses: fylein/python-pytest-github-action@v2
+        with:
+          args: pip3 install -r requirements.txt && pytest
```

Comment:
Take a look at [this example](https://github.com/pavelzw/pytest-action?tab=readme-ov-file#example-usage). Maybe you can do something similar?

Reply:
Done in [c731f16](https://github.com/InPreD/samplesheet_generator/pull/1/commits/c731f16a6ec275bff27a340b2fbaf6559f67be2e)